### ISSUE TITLE
Youtube API sometimes breaks for a tab if the page is still loading when a video is sent to min-vid

### ISSUE THREAD

**Issue by pdehaan** [meandavejustice/min-vid#328]
Found in latest 0.2.0.

Occasionally after watching a handful of videos from https://www.reddit.com/r/videos, I've noticed that eventually videos stop loading into the mini player.
Closing the mini player and relaunching the videos from the context menu seems to get the videos to correctly load.

I haven't seen any concrete errors in the Browser Console (mainly because of a constant flood of errors/warnings in the console), but this one looks a bit suspect:

![browser_console](https://cloud.githubusercontent.com/assets/557895/18890348/8daab496-84b5-11e6-82fa-7fd03a340dbf.png)
**Figure 1:** `TypeError: window.YTPlayer.getDuration is not a function` [bundle.js:120:15](https://github.com/meandavejustice/min-vid/blob/ca7c6634e598bf76edfb7836f1017425082da7a8/client-lib/yt-ctrl.js#L64).

**Comment by meandavejustice** [comment]
This is a dupe of #275 But since this includes the error and is fresher I'm going to close #275

**Comment by meandavejustice** [comment]
I think this has to do with a weird edge case where a video is loading, and you send another video. This doesn't always fail, but occasionally. I think it has something to do with `onLoaded` right when the new `src` is sent to the video. The fix is probably in clearing out the `window.YTPlayer` object if the player is loading and it just got a new `src`. This should probably be done in a `componentWillUpdate` or similar react lifecycle event.

**Comment by 6a68** [comment]
I updated the title, thinking I had consistent steps to reproduce. However, after closing Firefox, then running through the steps again, this only sometimes works:

Updating the title because I think I have consistent steps.

1. Open a new window, go to youtube.com, send a video to the min-vid player.
2. Open a second tab in that window, open up r/videos, and *very quickly, before the page has settled*, right-click a link and send to player.
3. The player is now broken for that second tab:
- the loading screen will be shown forever
- for that tab, other youtube links in the page seem not to work (but sometimes they do...) , but vimeo will load just fine always
- in rare cases, when the player is stalled on the loading screen, if I try to send a youtube video, the controls totally disappear, except for the drag handle. The player can't be closed and won't accept any new videos from any tab. (This level of brokenness is rare and hard to reproduce, and the simple workaround is to close the tab/window to get rid of the player.)

I have been able to pretty consistently repro this bug in two ways: 
- quickly load a new tab and send a video to min-vid before the page is fully loaded
- load two video websites in two tabs, let the pages settle, then quickly: switch tabs and send a video to min-vid
